### PR TITLE
Added Support For Token Provider

### PR DESCRIPTION
**Pull request by pbalexlear**
Added support for Token provider as an authentication method against service bus

### CODE REVIEW

File: Pat.Sender/MessageSender.cs [hunk 1]
```diff
@@ -42,7 +43,15 @@ public async Task SendMessages(IEnumerable<Message> messages)
                 var connectionString = _connectionResolver.GetConnection();
                 try
                 {
-                    var client = TopicClientResolver.GetTopic(connectionString, _senderSettings.EffectiveTopicName);
+                    TopicClient client ;
+                    if (_senderSettings.TokenProvider != null) {
```
Comment (Bigtalljosh) [inline Pat.Sender/MessageSender.cs]: Sorry, pedantic one. Can we keep the opening brace on a new line to keep consistency please :)

File: Pat.Sender/MessageSender.cs [hunk 2]
```diff
@@ -42,7 +43,15 @@ public async Task SendMessages(IEnumerable<Message> messages)
...
+                        client = TopicClientResolver.GetTopic(connectionString, _senderSettings.EffectiveTopicName, _senderSettings.TokenProvider);
+                    }
+                    else
+                    {
+                        client = TopicClientResolver.GetTopic(connectionString, _senderSettings.EffectiveTopicName);
+
```
Comment (Bigtalljosh) [inline Pat.Sender/MessageSender.cs]: Also remove the whitespace please

File: Pat.Sender/TopicClientResolver.cs [hunk 3]
```diff
@@ -28,5 +30,25 @@ public static TopicClient GetTopic(string connectionString, string topicName)
...
+            }
+
+            lock (Locks.GetOrAdd(connectionString, new object()))
+            {
+                if (TokenClients.ContainsKey(connectionString))
+                {
+                    return TokenClients[connectionString];
+                }
+                ServiceBusConnectionStringBuilder builder = new ServiceBusConnectionStringBuilder(connectionString);
+                var topicClient = new TopicClient(builder.Endpoint, topicName, tokenProvider,retryPolicy: RetryPolicy.Default);
```
Comment (Bigtalljosh) [inline Pat.Sender/TopicClientResolver.cs]: Formatting, if you press ctrl + K, ctrl + D in VS it'll format the whole file for you

File: Pat.Sender/MessageSender.cs (hunk 2 above)
Comment (pbalexlear) [reply Pat.Sender/MessageSender.cs]: fixed

**Comment by Bigtalljosh** [comment]
Given we're changing the framework version, I think we need to bump the major version number in the `appveyor.yml` as this will be a breaking change.

**Pull request merged**